save-agent/src/linuxX64Main/kotlin/com/saveourtool/save/agent/SaveAgent.kt

```diff
@@ -158,9 +158,7 @@ class SaveAgent(internal val config: AgentConfiguration,
                 state.value = AgentState.CLI_FAILED
             } else {
                 handleSuccessfulExit().invokeOnCompletion { cause ->
-                    if (cause == null) {
-                        state.value = AgentState.FINISHED
-                    }
+                    state.value = if (cause == null) AgentState.FINISHED else AgentState.CLI_FAILED
```

Comment:
I think a more suitable status here would be `CRASHED` instead of `CLI_FAILED`, because at this point we know that save-cli exited with status 0, but save-agent failed to save some of the execution data inside `handleSuccessfulExit`.

Reply:
fixed